telegram/files/file.py

```diff
@@ -111,33 +111,39 @@ def download(
               the path of a local file (which is the case when a Bot API Server is running in
               local mode), this method will just return the path.
 
+        .. versionchanged:: 14.0
+            * :param:`custom_path` now also accepts pathlib.Path objects as argument.
+            * Returns `pathlib.Path` object in cases where previously returned `str` object.
+
         Args:
-            custom_path (:obj: 'Path' | :obj:`str`, optional): Custom path.
+            custom_path (:obj: 'pathlib.Path' | :obj:`str`, optional): Custom path.
             out (:obj:`io.BufferedWriter`, optional): A file-like object. Must be opened for
                 writing in binary mode, if applicable.
             timeout (:obj:`int` | :obj:`float`, optional): If this value is specified, use it as
                 the read timeout from the server (instead of the one specified during creation of
                 the connection pool).
 
         Returns:
-            :obj:`str` | :obj:`io.BufferedWriter`: The same object as :attr:`out` if specified.
+            :obj:`pathlib.Path` | :obj:`io.BufferedWriter`: The same object as :attr:`out` if
+                specified.
             Otherwise, returns the filename downloaded to or the file path of the local file.
 
         Raises:
             ValueError: If both :attr:`custom_path` and :attr:`out` are passed.
 
         """
         if custom_path is not None and out is not None:
-            raise ValueError('custom_path and out are mutually exclusive')
+            raise ValueError('`custom_path` and `out` are mutually exclusive')
 
         local_file: bool = is_local_file(self.file_path)
-        url: Union[Path, str] = Path(self.file_path) if local_file else self._get_encoded_url()
+        url: Optional[str] = None if local_file else self._get_encoded_url()
+        path: Optional[Path] = Path(self.file_path) if local_file else None
```

Comment:
here again a type checker can infere the type automatically: `None` is obvious and the return value of `Path()` and `self._get_encoded_url` can be easily inferred. Please remove the annotations ;)

Reply:
done. also removed type annotation from the ```local_file``` variable.